mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/mesos/MesosDriverSupplier.java

```diff
@@ -76,10 +92,31 @@ public MesosSchedulerDriver get() {
                     .setCheckpoint(true)
                     .build();
             logger.info("initializing mesos scheduler driver");
-            final MesosSchedulerDriver mesosDriver =
-                    new MesosSchedulerDriver(mesosSchedulerCallbackHandler, framework, masterConfig.getMasterLocation());
+            MesosSchedulerDriver mesosDriver;
+            try {
+                mesosDriver = initMesosSchedulerDriverWithTimeout(mesosSchedulerCallbackHandler, framework, 5);
+            } catch (Exception e) {
+                logger.warn("timed out trying to initialize MesosSchedulerDriver, will retry", e);
+                isInitialized.compareAndSet(true, false);
+                mesosDriver = get();
+            }
+
             boolean result = mesosDriverRef.compareAndSet(null, mesosDriver);
             logger.info("initialized mesos scheduler driver {}", result);
+        } else {
+            // block till mesosDriver is not null
+            while (true) {
```

Comment:
Minor thing, maybe something like this to make it easier to read
```java
while (mesosDriverRef.get() == null) {
//wait
}
return mesosDriverRef.get();
```
Don't even need the else block here.